src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/RazorHoverProvider.ts

```diff
@@ -62,20 +64,36 @@ export class RazorHoverProvider
             // string.
 
             if (typeof content === 'string') {
-                rewrittenContent.push(content);
+                const filteredValue = this.filterGeneratedNamespace(content);
+                rewrittenContent.push(filteredValue);
             } else if ((content as { language: string; value: string }).language) {
                 const contentObject = (content as { language: string; value: string });
                 const markdownString = new vscode.MarkdownString();
-                markdownString.appendCodeblock(contentObject.value, contentObject.language);
+                const filteredValue = this.filterGeneratedNamespace(contentObject.value);
+                markdownString.appendCodeblock(filteredValue, contentObject.language);
                 rewrittenContent.push(markdownString);
             } else {
                 const contentValue = (content as vscode.MarkdownString).value;
-                const markdownString = new vscode.MarkdownString(contentValue);
+                const filteredValue = this.filterGeneratedNamespace(contentValue);
+                const markdownString = new vscode.MarkdownString(filteredValue);
                 rewrittenContent.push(markdownString);
             }
         }
 
         const hover = new vscode.Hover(rewrittenContent, remappedResponse.ranges[0]);
         return hover;
     }
+
+    private filterGeneratedNamespace(content: string): string {
+        // May contain the generated namespace which needs to be filtered out
+        /* Example:
+            ```csharp\nint AspNetCore_325c1543a2e5ffb7c4b17d99519708c8ca60e845.currentCount\n```\n
+        */
+
+        if (content == null || content.length <= RazorHoverProvider.GENERATED_PREFIX_LENGTH) {
+            return content;
+        }
+
+        return content.replace(/AspNetCore_\w{40}\./g, '');
```

Comment:
I know that any time we use a Regex without setting a timeout in C# we invoke the ire of @blowdart, and I suspect it's the same kind of deal in Typescript but I didn't find a way to set a timeout for a TypeScript RegEx,  so I'm not sure.

Reply:
Well that seems rather dumb of typescript doesn't it?

This ends up compiled to javascript that runs client side doesn't it? So if someone attacks it they're, well, attacking themselves.

Reply:
Oh it appears javascript has no such ability anyway. So, if this is running client side, it's fine. Just bleurgh,